projects/igniteui-angular/schematics/utils/dependency-handler.ts

```diff
@@ -59,32 +44,72 @@ export function addDependencies(options: Options): Rule {
     };
 }
 
-function addHammerJsToWorkspace(tree: Tree): Tree {
-    try {
-        const workspace = getWorkspace(tree);
-        const addedtoBuildScripts = addHammerToAngularWorkspace(workspace, 'build');
-        const addedtoToTestScripts = addHammerToAngularWorkspace(workspace, 'test');
+/**
+ * Recursively search for the first property that matches targetProp within the angular.json file.
+ */
+export function getPropertyFromWorkspace(targetProp: string, workspace: any, curKey = ''): any {
+    if (workspace.hasOwnProperty(targetProp)) {
+        return { key: targetProp, value: workspace[targetProp] };
+    }
 
-        if (addedtoBuildScripts || addedtoToTestScripts) {
-            overwriteJsonFile(tree, 'angular.json', workspace);
+    const workspaceKeys = Object.keys(workspace);
+    for (const key of workspaceKeys) {
+        // If the target property is an array, return its key and its contents.
+        if (Array.isArray(workspace[key])) {
+            return {
+                key: curKey,
+                value: workspace[key]
+            };
+        } else if (workspace[key] instanceof Object) {
+            // If the target property is an object, go one level in.
+            if (workspace.hasOwnProperty(key)) {
+                const newValue = getPropertyFromWorkspace(targetProp, workspace[key], key);
+                if (newValue !== null) {
+                    return newValue;
+                }
+            }
         }
+    }
 
-        return tree;
-    } catch (e) {
-        if (e.toString().includes('Could not find (undefined)')) {
-            throw new SchematicsException('angular.json was not found in the project\'s root');
-        }
+    return null;
+}
 
-        throw new Error(e.message);
-    }
+function includeDependencies(pkgJson: any, context: SchematicContext, tree: Tree, dependenciesKey: string) {
+    Object.keys(pkgJson.dependencies).forEach(pkg => {
+        const version = pkgJson.dependencies[pkg];
+        switch (pkg) {
+            case 'hammerjs':
+                logIncludingDependency(context, pkg, version);
+                addPackageToPkgJson(tree, pkg, version, dependenciesKey);
+
+                const workspace = getWorkspace(tree);
+                const targetProp = getPropertyFromWorkspace('scripts', workspace);
+                const mainTsPath = 'src/main.ts';
```

Comment:
'src/main.ts' should come from the workspace I think (safer). Something like a combo of the default project source root + entry